pkg/apis/ml/v1alpha1/storage_spec.go

```diff
@@ -21,47 +21,30 @@
 package v1alpha1
 
 import (
-	"github.com/pkg/errors"
-	core "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
+	"github.com/arangodb/kube-arangodb/pkg/apis/shared"
 )
 
 type ArangoMLStorageSpec struct {
-	// ListenPort defines on which port the sidecar container will be listening for connections
-	// +doc/default: 9201
-	ListenPort *uint16 `json:"listenPort,omitempty"`
-
-	// Resources holds resource requests & limits for container running the S3 proxy
-	// +doc/type: core.ResourceRequirements
-	// +doc/link: Documentation of core.ResourceRequirements|https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#resourcerequirements-v1-core
-	Resources core.ResourceRequirements `json:"resources,omitempty"`
-
-	S3 *ArangoMLStorageS3Spec `json:"s3,omitempty"`
+	Mode    *ArangoMLStorageSpecMode    `json:"mode,omitempty"`
```

Comment:
GetMode + Documentation missing

Reply:
Didn't add docs because they are not rendered anyway. 
But guess it is still nice to have for programmatic use. Added.